sys/net/application_layer/nanocoap/sock.c

```diff
@@ -68,8 +82,13 @@ ssize_t nanocoap_request(coap_pkt_t *pkt, sock_udp_ep_t *local, sock_udp_ep_t *r
                 tries_left--;
                 if (!tries_left) {
                     DEBUG("nanocoap: maximum retries reached\n");
+                    break;
+                }
+                else {
+                    timeout *= 2;
+                    res = -EAGAIN;
+                    continue;
```

Comment:
Is it intentional to multiply the timeout with 2 twice?

Reply:
I kept the [pre-existing logic](https://github.com/RIOT-OS/RIOT/blob/master/sys/net/application_layer/nanocoap/sock.c#L67) here, timeout is doubled with each retry.

Reply:
I was just asking because in line 81 the timeout is also doubled (the code line you've linked is unchanged), if this is correct then you can simply resolve this comment :)

Reply:
Ah good catch!